src/pages/Post/PostDetail/hooks/useMessagesPage.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * 상태 및 로직만 담당
+ */
+import { useCallback, useState } from "react";
+import { useNavigate, useMatch, useParams } from "react-router-dom";
+import useInfiniteMessages from "./useInfiniteMessages";
+import useModal from "../../../../components/Modal/useModal";
+import { deleteMessages } from "../../../../api/delete/deleteMessages";
+
+const useMessagesPage = () => {
+  const { id: recipientId } = useParams();
+  const navigate = useNavigate();
+  const editMode = !!useMatch("/post/:id/edit");
+  const [selectedIds, setSelectedIds] = useState([]);
+  const { showModal } = useModal();
+  const {
+    messages,
+    setMessages,
+    fetchMore,
+    isLast,
+    error: fetchError,
+  } = useInfiniteMessages({ id: recipientId, limit: editMode ? 6 : 5 });
+  const [error, setError] = useState("");
+
+  const handleSave = async () => {
+    try {
+      await Promise.all(selectedIds.map((id) => deleteMessages({ id })));
+      setMessages((prev) =>
+        prev.filter((msg) => !selectedIds.includes(msg.id))
+      );
+      setSelectedIds([]);
+      navigate(`/post/${recipientId}`);
+    } catch (err) {
+      console.error("삭제 실패", err);
+      setError("메시지 삭제 중 문제가 발생했습니다.");
+    }
+  };
+
+  const handleEditButton = () => {
+    const baseUrl = `/post/${recipientId}`;
+    setSelectedIds([]);
+    navigate(editMode ? baseUrl : `${baseUrl}/edit`);
+  };
+
+  const toggleSelection = useCallback((id) => {
+    setSelectedIds((prev) =>
+      prev.includes(id) ? prev.filter((x) => x !== id) : [...prev, id]
+    );
+  }, []);
+
+  const handleDeleteSelected = async () => {
+    try {
+      await Promise.all(selectedIds.map((id) => deleteMessages({ id })));
+      setMessages((prev) =>
+        prev.filter((msg) => !selectedIds.includes(msg.id))
+      );
+      setSelectedIds([]);
+    } catch (err) {
+      console.error("삭제 실패", err);
+      setError("메시지 삭제 중 문제가 발생했습니다.");
+    }
+  };
+
+  const handleToggleSelectAll = () => {
+    const allIds = messages.map((m) => m.id);
+    if (selectedIds.length === allIds.length) {
```

Comment:
이 부분때문에 영상에서처럼 의도대로 동작하지 않는 것 같습니다! 
```suggestion
    if (selectedIds.length > 0) {
```
이렇게 고치면 하나라도 선택되어 있을 때 전체해제가 될 것 같습니다!

의도하신게 모두 선택되어있을때만 전체 해제를 원하시는 거라면 버튼의 이름을 수정하셔야 할 것 같습니다!

Reply:
놓치고 있던 부분인데 알려주셔서 감사합니다!